daemon/algod/api/server/v2/dryrun.go

```diff
@@ -397,17 +397,22 @@ func doDryrunRequest(dr *DryrunRequest, response *generated.DryrunResponse) {
 		return
 	}
 	proto := config.Consensus[protocol.ConsensusVersion(dr.ProtocolVersion)]
-
+	proto.EnableAppCostPooling = true
+	proto.MaxTxGroupSize = 16
```

Comment:
For MaxTxGroupSize, we should figure out why it's not set.  That has been set for some time, so I don't see why you'd want it set here.

EnableCostPoooling is more interesting - On the one hand, if it's not set, the user may want his 701 unit app call to fail so that they can understand why.  But another way of looking at this is that you're seting EnableCostPooling unconditionally (and "locally") so that you can report the cost properly, and the user will thank you for reporting the 701 cost. Perhaps the best behavior is to
 1. Record the original value of EnableCostPooling
 2. Set it here so that the dry run can run long.
 3. If the call passes, but cost exceeds MaxAppCost and EnableCostPooling was not originally on, make it a failure.
 4. Report the cost either way.

Reply:
You could also, if we want to make the dry run have higher fidelity output, also turn success into failure if the costs exceeds a tighter bound on the actual cost that would be available.  You could calculate that, in increasingly correct ways.
1. Use the size of the submitted request.
2. Use the number of app calls in the submitted request.
3. Actually track the available budget that would be available as multiple txns run.